src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -690,6 +715,216 @@ internal Region CreateRegion(int startColumn, int startLine, int endColumn = 0,
             return region;
         }
 
+        public int GetRegionCharLength(RegionVersionOne v1Region, Uri uri)
+        {
+            int result = 0;
+
+            TextReader reader = null;
+            Stream stream = null;
+
+            try
+            {
+                // Get a TextReader so we can read line by line
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                if (reader != null)
+                {
+                    // Read each line up to the startLine and copy them into a StringBuilder
+                    StringBuilder sb = new StringBuilder();
+                    for (int i = 1; i < v1Region.StartLine; sb.AppendLine(reader.ReadLine()), i++) { }
+
+                    // Read the startLine and append up to the startColumn
+                    string line = reader.ReadLine();
+                    sb.Append(line.Substring(0, v1Region.StartColumn));
+
+                    // Calculate the byte size of the content and close the reader
+                    int byteOffset = SarifUtilities.GetByteLength(sb.ToString(), encoding);
+                    reader.Close();
+
+                    // Get a stream so we can do byte operations
+                    stream = GetContentStream(uri);
+
+                    if (stream != null)
+                    {
+                        // Seek to the start of the region
+                        stream.Seek(byteOffset, SeekOrigin.Begin);
+
+                        // Read the region's bytes
+                        byte[] bytes = new byte[v1Region.Length];
+                        stream.Read(bytes, 0, bytes.Length);
+
+                        // Calculate the character length of the region
+                        string s = encoding.GetString(bytes);
+                        result = s.Length;
+                    }
+                }
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+
+                if (stream != null)
+                {
+                    stream.Dispose();
+                }
+            }
+
+            return result;
+        }
+
+        public int GetRegionCharOffset(RegionVersionOne v1Region, Uri uri)
+        {
+            int result = 0;
+
+            TextReader reader = null;
+            Stream stream = null;
+
+            try
+            {
+                // Get a reader so the encoding might be detected by the StreamReader ctor
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                if (reader != null)
+                {
+                    // Close the reader
+                    reader.Close();
+
+                    // Get a stream so we can do byte operations
+                    stream = GetContentStream(uri);
```

Comment:
Now... if `reader` is a `StreamReader`, you just have to do `stream = reader.BaseStream;` which is more efficient.

Reply:
Done